src/node/mod.rs

```diff
@@ -0,0 +1,410 @@
+//! Module for abstractions on drm device nodes.
+
+pub mod constants;
+
+use std::error::Error;
+use std::fmt::{self, Display, Formatter};
+use std::io;
+use std::os::unix::io::AsFd;
+use std::path::{Path, PathBuf};
+
+use rustix::fs::{fstat, major, minor, stat, Dev as dev_t, Stat};
+
+use crate::node::constants::*;
+
+/// A node which refers to a DRM device.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct DrmNode {
+    dev: dev_t,
+    ty: NodeType,
+}
+
+impl DrmNode {
+    /// Creates a DRM node from an open drm device.
+    ///
+    /// This function does not take ownership of the passed in file descriptor.
```

Comment:
Doesn't `AsFd` already signify that? (Though one could both move or borrow an `OwnedFd` with this API, IIRC)

Reply:
I don't see much harm in clarifying either way.

Reply:
The comment is a relic from before https://github.com/Smithay/smithay/commit/b098caa4579 when it did take a `RawFd`.

Reply:
Seeing as the same lifetime comment wasn't on the `A: AsRawFd` function there, I think we're good to remove it as this is already communicated by the API.